src/librustc_metadata/encoder.rs

```diff
@@ -668,14 +668,15 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
             hir::ItemDefaultImpl(..) => {
                 let data = ImplData {
                     polarity: hir::ImplPolarity::Positive,
+                    defaultness: hir::Defaultness::Final,
```

Comment:
@nagisa I've read the discussion about `UnwindSafe` https://github.com/rust-lang/rust/issues/40628, this PR however is one of a set of tasks described here https://github.com/rust-lang/rust/issues/37653, 
the 3rd 

> all items in a default impl are (implicitly) default and hence specializable

It doesn't address default trait implementations.
At least another PR must follow in order to cover the other two points.
so we could define better what @abonander  says about 

> re-addressed this in the context of specialization

and add it to the task list.
What do you think?